dask/array/routines.py

```diff
@@ -523,34 +523,44 @@ def gradient(f, *varargs, **kwargs):
     return results
 
 
+def _bincount_sum(bincounts, dtype=int):
+    n = max(map(len, bincounts))
+    out = np.zeros(n, dtype=dtype)
+    for b in bincounts:
+        out[:len(b)] += b[:len(b)]
+    return out
+
+
 @wraps(np.bincount)
-def bincount(x, weights=None, minlength=None):
-    if minlength is None:
-        raise TypeError("Must specify minlength argument in da.bincount")
+def bincount(x, weights=None, minlength=0):
     assert x.ndim == 1
     if weights is not None:
         assert weights.chunks == x.chunks
 
     # Call np.bincount on each block, possibly with weights
+    dsk = {}
     token = tokenize(x, weights, minlength)
-    name = 'bincount-' + token
-    if weights is not None:
-        dsk = {(name, i): (np.bincount, (x.name, i), (weights.name, i), minlength)
-               for i, _ in enumerate(x.__dask_keys__())}
-        dtype = np.bincount([1], weights=[1]).dtype
-    else:
-        dsk = {(name, i): (np.bincount, (x.name, i), None, minlength)
-               for i, _ in enumerate(x.__dask_keys__())}
-        dtype = np.bincount([]).dtype
+    for i, _ in enumerate(x.__dask_keys__()):
+        if weights is not None:
+            dsk[('bincount-' + token, i)] = (np.bincount, (x.name, i),
+                                             (weights.name, i), minlength)
+            dtype = np.bincount([1], weights=[1]).dtype
+        else:
+            dsk[('bincount-' + token, i)] = (np.bincount, (x.name, i),
+                                             None, minlength)
+            dtype = np.bincount([]).dtype
 
-    # Sum up all of the intermediate bincounts per block
+    bincount_list = [i for i in list(dsk) if i[0] == 'bincount-' + token]
     name = 'bincount-sum-' + token
```

Comment:
Maybe we can move the definition of `name` up by `tokenize` again (maybe even removing the `token` intermediate variable)

Reply:
Sure. I don't really understand what what token is doing here, tbh

Reply:
`token` variable has been removed.

Reply:
Ah, sorry, the tokenize process is still improtant.  Every task is uniquely named in Dask, similar to git commits, to help with deduplication.  I've reverted this and changed things a little bit in https://github.com/dask/dask/pull/4684/commits/41c6b00f96a3c0c11a6d374ce9500ee24fa912c3 and added tests to show why it's used.

Reply:
Aaah, ok. I had assumed that was what it was for, but then got confused about what you were trying to say with this comment. Thanks for clarifying!